packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -87,4 +103,19 @@ export class DefaultHasOneRepository<
     }
     return found[0];
   }
+  async delete(options?: Options): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return targetRepository.deleteAll(options);
```

Comment:
This will delete ALL target model instances, not only the instance associated with the source model instance.

Please add a test to verify that `delete` preserves instances that are not related to the source model and then fix the implementation.

I think this is something we should verify for the PATCH operation too. Please add another test to check that PATCH is updating only the single instance on the other end of the relation. 

Yet another two tests to consider: what happens when there is no target model instance? IMO, both `delete` and `patch` should run successfully and return `{count: 0}`.

Reply:
Hi,
I did what you requested to remove the where argument etc... Note that I implemented hasOne same as hasMany is implemented and in hasMany there is no any test to verify the associated target model, the relation already handle it. Please see how hasMany is implement. If your assumption is right the same issue is existing in hasMany. But, when I made tests in hasOne with my initial code including PUT, hasOne worked as expected.
Just to keep you informed, someone else will continue the updates of hasOne with you.
I am moving to other responsibilities.
Kind regards
Raphael

Reply:
I am afraid we misunderstood each other. I was asking to remove `where` argument from the public API, not from the underlying implementation...

> Just to keep you informed, someone else will continue the updates of hasOne with you.
I am moving to other responsibilities.

Thank you for the update, I wish you good luck on your new assignment 🍀